test/models/user_test.rb

```diff
@@ -4,4 +4,35 @@ class UserTest < ActiveSupport::TestCase
   # test "the truth" do
   #   assert true
   # end
+
+  EMAIL_REGEX = /\A[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}\Z/i
+
+  def setup
+    @user = User.create(username: "User1", email: "user1sfu.ca", password: "secrete")
+  end
+
+  test "should not have the same username" do
+    duplicate_user = @user.dup
```

Comment:
in these tests, @user and duplicate_user will have the same username as well as email. 

Should have 1 test case where only the username is duplicate, and 1 test case where only the email is duplicate, and test that the user should be invalid at both of these points?